plugins/labormanager.cpp

```diff
@@ -673,8 +673,15 @@ static df::building* get_building_from_job(df::job* j)
     return 0;
 }
 
-static df::unit_labor construction_build_labor (df::item* i)
+static df::unit_labor construction_build_labor (df::building_actual* b)
 {
+    if (b->getType() == df::building_type::RoadPaved)
+        return df::unit_labor::BUILD_ROAD;
+    auto a = (df::building_actual *) b;
+    // For screw pumps contained_items[0] = pipe, 1 corkscrew, 2 block
+    // For wells 0 mechanism, 1 rope, 2 bucket, 3 block
+    // Trade depots and bridges use the last one too
+    df::item* i = a->contained_items.back()->item;
```

Comment:
A constructed building will always have at least one item in it. I doubt that's anything that will change any time soon.